go/multipooler/manager/manager.go

```diff
@@ -141,14 +142,30 @@ type cachedMultiPoolerInfo struct {
 }
 
 // NewMultiPoolerManager creates a new MultiPoolerManager instance
-func NewMultiPoolerManager(logger *slog.Logger, config *Config) *MultiPoolerManager {
+func NewMultiPoolerManager(logger *slog.Logger, config *Config) (*MultiPoolerManager, error) {
 	return NewMultiPoolerManagerWithTimeout(logger, config, 5*time.Minute)
 }
 
 // NewMultiPoolerManagerWithTimeout creates a new MultiPoolerManager instance with a custom load timeout
-func NewMultiPoolerManagerWithTimeout(logger *slog.Logger, config *Config, loadTimeout time.Duration) *MultiPoolerManager {
+func NewMultiPoolerManagerWithTimeout(logger *slog.Logger, config *Config, loadTimeout time.Duration) (*MultiPoolerManager, error) {
 	ctx, cancel := context.WithCancel(context.TODO())
```

Comment:
Seems cleaner to move the context.WithCancel() here below the validation so it's not necessary to remember to call cancel() manually in each validation error case?